### ISSUE TITLE
When loading test hook from github, do not assume 4-2-stable

### ISSUE THREAD

**Issue by d-w-moore** [irods/irods_testing_environment#79]
https://github.com/irods/irods_testing_environment/blob/1198efdf56f813fc45a12f0f502e9693ad638b26/run_plugin_tests.py#L111

When not providing the test hook via the filesystem path, do not assume certain fixed values like the branch name. This applies especially now that we now, have only the one unified git branch for the irods testing environment.

I've discussed what to do, very briefly, with @alanking ... I think our consensus was along the lines that we need to be able to inject an entire dictionary of appropriate values into that level of the code, not just the branch-name. It just remains how to implement that effectively on the command line.

**Comment by alanking** [comment]
> It just remains how to implement that effectively on the command line.

Agreed. The plugin test runner script might be worth reworking at some point. The options provided are supposed to allow for a high level of flexibility but it's not comprehensive (read: this issue :)) and can sometimes be confusing in purpose if the user is not familiar with the implementation.

And for reference, here is the signature and docstring for the referenced function:
https://github.com/irods/irods_testing_environment/blob/1198efdf56f813fc45a12f0f502e9693ad638b26/irods_testing_environment/test_utils.py#L118-L126
